baseapp/abci_test.go

```diff
@@ -1998,22 +2001,50 @@ func TestABCI_HaltChain(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			suite := NewBaseAppSuite(t, baseapp.SetHaltHeight(tc.haltHeight), baseapp.SetHaltTime(tc.haltTime))
-			suite.baseApp.InitChain(&abci.RequestInitChain{
-				ConsensusParams: &cmtproto.ConsensusParams{},
-				InitialHeight:   tc.blockHeight,
-			})
-
-			app := suite.baseApp
-			_, err := app.FinalizeBlock(&abci.RequestFinalizeBlock{
-				Height: tc.blockHeight,
-				Time:   time.Unix(tc.blockTime, 0),
-			})
+			// Set up signal handling before the test
+			sigCh := make(chan os.Signal, 1)
+			signal.Notify(sigCh, os.Interrupt, syscall.SIGTERM)
+			defer signal.Stop(sigCh)
+
+			// Run in a goroutine to handle the signal
+			done := make(chan struct{})
+			var finalizeErr error
+
+			go func() {
+				defer close(done)
+
+				suite := NewBaseAppSuite(t, baseapp.SetHaltHeight(tc.haltHeight), baseapp.SetHaltTime(tc.haltTime))
+				suite.baseApp.InitChain(&abci.RequestInitChain{
+					ConsensusParams: &cmtproto.ConsensusParams{},
+					InitialHeight:   tc.blockHeight,
+				})
+
+				app := suite.baseApp
+				_, finalizeErr = app.FinalizeBlock(&abci.RequestFinalizeBlock{
+					Height: tc.blockHeight,
+					Time:   time.Unix(tc.blockTime, 0),
+				})
+			}()
+
+			if tc.expHalt {
+				// Wait for the signal
+				select {
+				case sig := <-sigCh:
+					t.Logf("Received expected signal: %v", sig)
+				case <-time.After(1 * time.Second):
```

Comment:
This seems a little too strict of a timeout.  Maybe make it longer (like 5s or so)?